### PR TITLE
Test for exhaustiveness cut-off.

### PR DESCRIPTION
**Pull request by davidungar**
Adds a test for the recently-added -switch-checking-invocation-threshold= frontend option.

### CODE REVIEW

**Comment by davidungar** [comment]
@swift-ci please smoke test

File: test/Sema/exhaustive_switch_invocation_threshold.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// Ensure that -switch-checking-invocation-threshold= frontend option works.
+//
+// RUN: not %target-swift-frontend -typecheck %s -switch-checking-invocation-threshold=1 2>&1 | %FileCheck %s --check-prefix LIMIT
+// RUN: not %target-swift-frontend -typecheck %s  2>&1 | %FileCheck %s --check-prefix NO_LIMIT
+// LIMIT: error: the compiler is unable to check that this switch is exhaustive in reasonable time
+// LIMIT-NOT: error: switch must be exhaustive
+// NO_LIMIT: error: switch must be exhaustive
+// NO_LIMIT-NOT: error: the compiler is unable to check that this switch is exhaustive in reasonable time
```
Comment (jrose-apple) [inline test/Sema/exhaustive_switch_invocation_threshold.swift]: This looks fine, but it's worth pointing out that `-NOT` lines only check that certain text does not appear after the previous positive check and before the next positive check. You could either bracket your `LIMIT`/`NO_LIMIT` lines with the `-NOT` checks, or save the error output to a file and check it twice (a common enough idiom in our test suite, look for `-check-prefix NEGATIVE`), or just not worry too much about it.

**Review by jrose-apple — APPROVED** [review]

**Comment by davidungar** [comment]
@swift-ci please smoke test

**Comment by davidungar** [comment]
@swift-ci please smoke test

**Review by jrose-apple — APPROVED** [review]
This works too. It's not the way we'd usually do negative tests, but I like that it's symmetrical.

**Comment by davidungar** [comment]
Thanks, Jordan.

**Pull request merged**